src/graph_building/graph_data_generator.py

```diff
@@ -1,9 +1,100 @@
+import pandas as pd
+
 from src.data.interfaces.data_interface import DataInterface
+from src.graph_building.interfaces.vertex_data_interface import VertexDataInterface
 
 
 class GraphDataGenerator:
     """
     This class finds all vertices and edges of the FWG.
     """
-    def __init__(self, data_interface: DataInterface):
-        pass
+    def __init__(self,
+                 data_interface: DataInterface,
+                 beta: int,
+                 delta: int = 2
+                 ):
+        """
+        Constructor.
+        :param DataInterface data_interface: the DataInterface instance containing required data
+        :param int beta: the number of days allowed after a vertex for continuation
+        :param int delta: the number of days that a record is required to be greater
+                          than the records before, and to be greater or equal to after
+                          to be considered a peak
+        """
+        self.data_interface = data_interface
+
+        self.beta = beta
+        self.delta = delta
+
+        self.vertex_interface = VertexDataInterface()
+
+    def run(self):
+        """
+        Finds all vertices per gauge and connects them with edges.
+        """
+        gauges = self.data_interface.gauges
+
+        vertices = dict()
+        for gauge in gauges:
+            vertices[str(gauge)] = self.find_vertices(gauge=str(gauge))
+
+        river_kms = self.data_interface.meta['river_km'].tolist()
+
+        data = {
+            'vertices': vertices,
+            'river_kms': river_kms
+        }
+
+        self.vertex_interface = VertexDataInterface(data=data)
+
+    def find_vertices(self, gauge: str) -> dict:
```

Comment:
Ezt a metódust három metódus hívására lehet bontani:
- `get_series`
- `get_peaks`
- `get_peak_data`

Így már elég metódus van ahhoz, hogy a csúcskeresés külön osztályt kapjon.

A külön osztály `run` függvénye lehetne a `run` metódus LOC37-48 sorai. Ez az osztály maradjon, mert ez fogja majd össze a csúcs és élkeresést. Így az élkeresés már a `vertex_interface`-szel tud dolgozni. 

Az implementációs diagramot is módosítani kell ehhez.